scheduling/taskmetadata.go

```diff
@@ -28,18 +28,48 @@ type TaskMetadataManager struct {
 	idGenerator ffsync.OrderedIdGenerator
 }
 
-func NewMemoryTaskMetadataManager() TaskMetadataManager {
-	memoryLocker := ffsync.NewMemoryLocker()
-	memoryStorage := ss.NewMemoryStorageImplementation()
+func NewMemoryTaskMetadataManager() (TaskMetadataManager, error) {
+	memoryLocker, _ := ffsync.NewMemoryLocker()
+	memoryStorage, _ := ss.NewMemoryStorageImplementation()
 
 	memoryMetadataStorage := ss.MetadataStorage{
 		Locker:  &memoryLocker,
 		Storage: &memoryStorage,
 	}
+
+	idGenerator, _ := ffsync.NewMemoryOrderedIdGenerator()
+
 	return TaskMetadataManager{
 		storage:     memoryMetadataStorage,
-		idGenerator: ffsync.NewMemoryOrderedIdGenerator(),
+		idGenerator: idGenerator,
+	}, nil
+}
+
+func NewETCDTaskMetadataManager() (TaskMetadataManager, error) {
+	etcdLocker, err := ffsync.NewETCDLocker()
+	if err != nil {
+		return TaskMetadataManager{}, err
```

Comment:
since `TaskMetadataManager` is not interface, we can't return nil. I get the following warning: `cannot use nil as TaskMetadataManager value in return statement`

Reply:
ah good point. golang 😐